core/src/main/java/com/github/phantazmnetwork/core/gui/Gui.java

```diff
@@ -0,0 +1,286 @@
+        public @NotNull Builder withItem(@NotNull GuiItem item) {
+            items.add(Objects.requireNonNull(item, "item"));
+            if (items.size() > type.getSize()) {
+                throw new IllegalArgumentException("too many items for InventoryType " + type);
+            }
+
+            return this;
+        }
+
+        /**
+         * Adds a collection of items to the Gui. Will be positioned by the {@link SlotDistributor} when built. If there
+         * are more items than can be held by the Gui, throws an exception.
+         *
+         * @param items the items to add
+         * @return this instance, for chaining
+         */
+        public @NotNull Builder withItems(@NotNull Collection<? extends GuiItem> items) {
+            this.items.addAll(items);
+            if (items.size() > type.getSize()) {
+                throw new IllegalArgumentException("too many items for InventoryType " + type);
+            }
+
+            return this;
+        }
+
+        /**
+         * Builds the Gui, positioning items according to the {@link SlotDistributor} used by this builder.
+         *
+         * @return a new Gui
+         */
+        public @NotNull Gui build() {
+            int size = type.getSize();
+
+            int width = switch (type) {
+                case CHEST_1_ROW, CHEST_2_ROW, CHEST_3_ROW, CHEST_4_ROW, CHEST_5_ROW, CHEST_6_ROW, SHULKER_BOX -> 9;
+                case WINDOW_3X3 -> 3;
+                default -> size;
+            };
+
+            int height = size / width;
+
+            int[] slots = slotDistributor.distribute(width, height, items.size());
+
+            Gui gui = new Gui(type, title, dynamic);
+            for (int i = 0; i < slots.length; i++) {
+                gui.insertItem(items.get(i), slots[i]);
+            }
+
+            return gui;
+        }
+    }
+
+    /**
+     * Creates a new instance of the builder class which may be used to more easily construct Gui instances.
+     *
+     * @param inventoryType   the type of inventory used by the Gui
+     * @param slotDistributor the {@link SlotDistributor} used to position items
+     * @return the builder instance
+     */
+    public static @NotNull Builder builder(@NotNull InventoryType inventoryType,
+            @NotNull SlotDistributor slotDistributor) {
+        return new Builder(inventoryType, slotDistributor);
+    }
+
+    /**
+     * A {@link GuiItem} in a particular slot. Used by methods like {@link Gui#getItems()} to retain slot information.
+     *
+     * @param item the GuiItem
+     * @param slot the slot it's in
+     */
+    public record SlottedItem(@NotNull GuiItem item, int slot) {
+    }
+
+    private final GuiItem[] items;
+    private final boolean isDynamic;
+
+    /**
+     * Creates a new instance of this class.
+     *
+     * @param inventoryType the type of inventory to send
+     * @param title         the inventory's title text
+     * @param isDynamic     whether this GUI is "dynamic"
```

Comment:
define better

Reply:
?

Reply:
move the comment that says non dynamic GUIs don't update up to here or else you have to search in method bodies what dynamic means

Reply:
it's explained in the class javadoc